libimage/search.go

```diff
@@ -348,6 +331,41 @@ func searchRepositoryTags(ctx context.Context, sys *types.SystemContext, registr
 	return paramsArr, nil
 }
 
+// getDockerAuthConfig extracts a docker auth config from the CopyOptions.  Returns
+// nil if no credentials are set.
+func (options *SearchOptions) getDockerAuthConfig() (*types.DockerAuthConfig, error) {
```

Comment:
Per https://github.com/containers/common/pull/1328#discussion_r1109078458, I think it would be helpful for this function to fail if there are multiple credential sources — in some kind of symmetric way; now it detects `Username` + `Credentials`, but not the other ambiguities.